myhoard/controller.py

```diff
@@ -1168,9 +1179,16 @@ def _purge_old_binlogs(self, *, mysql_maybe_not_running=False):
                 last_binlog_purge=last_purge,
                 last_could_have_purged=last_could_have_purged,
             )
+
             self.stats.gauge_float("myhoard.binlog.time_since_any_purged", time.time() - last_purge)
             self.stats.gauge_float("myhoard.binlog.time_since_could_have_purged", time.time() - last_could_have_purged)
 
+            oldest_binlog_time = self._get_oldest_binlog_time()
+            self.stats.gauge_float(
+                "myhoard.binlog.time_since_oldest_should_have_purged", 
+                time.time() - (oldest_binlog_time + purge_settings["min_binlog_age_before_purge"])
```

Comment:
`time.time()` called three times here, could assign to a variable to only have it called once.

Reply:
done